src/lib/Microsoft.Health.Fhir.Ingest/Service/NormalizationStreamingEventConsumerService.cs

```diff
@@ -0,0 +1,119 @@
+
+using System;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Events.Common;
+using Microsoft.Health.Events.EventCheckpointing;
+using Microsoft.Health.Events.EventConsumers;
+using Microsoft.Health.Events.EventConsumers.Service;
+using Microsoft.Health.Events.Model;
+using Microsoft.Health.Fhir.Ingest.Data;
+using Microsoft.Health.Fhir.Ingest.Telemetry;
+using Microsoft.Health.Fhir.Ingest.Template;
+using Microsoft.Health.Logging.Telemetry;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class NormalizationStreamingEventConsumerService : StreamingEventConsumerService
+    {
+        private readonly IContentTemplate _contentTemplate;
+        private readonly IConverter<IEventMessage, JObject> _converter;
+        private readonly IExceptionTelemetryProcessor _exceptionTelemetryProcessor;
+        private readonly IEnumerableAsyncCollector<IMeasurement> _collector;
+
+        public NormalizationStreamingEventConsumerService(
+            EventBatchingOptions options,
+            ICheckpointClient checkpointClient,
+            ITelemetryLogger log,
+            IEventProcessingMetricMeters eventProcessingMetricMeters,
+            IConverter<IEventMessage, JObject> converter,
+            IContentTemplate contentTemplate,
+            IExceptionTelemetryProcessor exceptionTelemetryProcessor,
+            IEnumerableAsyncCollector<IMeasurement> collector)
+            : base(options, checkpointClient, log, eventProcessingMetricMeters)
+        {
+            _contentTemplate = EnsureArg.IsNotNull(contentTemplate, nameof(contentTemplate));
+            _converter = EnsureArg.IsNotNull(converter, nameof(converter));
+            _exceptionTelemetryProcessor = EnsureArg.IsNotNull(exceptionTelemetryProcessor, nameof(exceptionTelemetryProcessor));
+            _collector = EnsureArg.IsNotNull(collector, nameof(collector));
+        }
+
+        protected override async Task ConsumeEventImpl(IEventMessage eventArg)
+        {
+            try
+            {
+                RecordLatencyMetrics(eventArg);
+
+                var token = _converter.Convert(eventArg);
+
+                await NormalizeMessage(token, eventArg);
+            }
+            catch (Exception ex)
+            {
+                if (!_exceptionTelemetryProcessor.HandleException(ex, Logger))
+                {
+                    throw; // Immediately throw originalexception if it is not handled
+                }
+            }
+        }
+
+        private async Task NormalizeMessage(JToken token, IEventMessage eventArg)
+        {
+            try
+            {
+                int projectedMeasurements = 0;
+                foreach (var measurement in _contentTemplate.GetMeasurements(token))
+                {
+                    measurement.IngestionTimeUtc = eventArg.EnqueuedTime.UtcDateTime;
+                    await _collector.AddAsync(measurement);

```

Comment:
This will cause a single Measurement to be sent over to the internal EventHub at a time. In my testing it is faster to send over a batch of events versus a single event. 
An argument could be made that the `IEnumerableAsyncCollector` should handle the batching, and perhaps accept a `flush` call. But that would require a refactor.

Reply:
It should be done in the IAsync collector.  I will see what can be done there.

Reply:
Actually for now, I may just collect the events from the project and push those as a collection.  Does that match what is being done today?  Collecting and flushing across partitions will be more complicate and should be saved for a later refactor.

Reply:
Yes, collecting all projections from the DeviceEvent pushing them as a collection is what is done today. I think that a later refactor to deal with that batching in the collector makes sense.

Reply:
After moving the try/catch block inside the for loop that is harder to implement now but thanks for the suggestion.

I ended up doing a list that is reused for each message processed.  During processing, the generated measurements are stored and then sent as a batch to the async collector. Afterward the list is cleared of elements and reused for the next message.  The underlying array in the list will remain.  I set the initial capacity to 50 but it can grow beyond.

Long term, you are correct, I would like to make this the responsibility of the async collector.  I am creating a work item of additional tasks not covered by this PR and will include this.